internal/controller/persistentvolumeclaim_controller.go

```diff
@@ -66,16 +58,81 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 		}, nil
 	}
 
-	// Skip if the PVC is not deleted or PVC does not have TopoLVM's finalizer.
-	if pvc.DeletionTimestamp == nil {
+	// Skip if PVC does not have TopoLVM's finalizer.
+	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
 		return ctrl.Result{}, nil
 	}
 
-	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
-		return ctrl.Result{}, nil
+	if pvc.DeletionTimestamp == nil {
+		err := r.notifyKubeletToResizeFS(ctx, log, pvc)
+		return ctrl.Result{}, err
+	}
+
+	result, err := r.deletePodsUsingDeletingPVC(ctx, log, pvc)
+	if result.Requeue || err != nil {
+		return result, err
+	}
+
+	controllerutil.RemoveFinalizer(pvc, topolvm.PVCFinalizer)
+
+	if err := r.client.Update(ctx, pvc); err != nil {
+		log.Error(err, "failed to remove finalizer", "name", pvc.Name)
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *PersistentVolumeClaimReconciler) notifyKubeletToResizeFS(ctx context.Context, log logr.Logger, pvc *corev1.PersistentVolumeClaim) error {
+	// kubelet periodically checks pods to resize fs but it may take a long time.
+	// To speed up the process, mutate pods by adding an annotation.
+	// This process is only done when the PVC has `FileSystemResizePending` condition
+	// and pods are not mutated yet.
+	var condTransitionAt time.Time
+	for _, cond := range pvc.Status.Conditions {
+		if cond.Type == corev1.PersistentVolumeClaimFileSystemResizePending && cond.Status == corev1.ConditionTrue {
+			condTransitionAt = cond.LastTransitionTime.Time
+			goto needResizing
+		}
 	}
+	return nil
 
+needResizing:
+	pods, err := r.getPodsByPVC(ctx, pvc)
+	if err != nil {
+		log.Error(err, "unable to fetch PodList for a PVC", "pvc", pvc.Name, "namespace", pvc.Namespace)
```

Comment:
```suggestion
		log.Error(err, "unable to fetch PodList for a PVC")
```
the controller logger should already include a resource reference to the PVC because thats the object it reconciles